### ISSUE TITLE
Can't install under Lucee 5

### ISSUE THREAD

**Issue by Meikelv** [boughtonp/cfpassphrase#20]
I'm using Lucee 5.01.28.
I installed:
cfpassphrase.jar in: c:\lucee\tomcat\lucee-server\context\lib\
cfPassphrase.tld in: c:\lucee\tomcat\lucee-server\context\library\tld\
cfPassphrase.fld in: c:\lucee\tomcat\lucee-server\context\library\fld\

I received the followin error message:

Lucee 5.0.1.28-SNAPSHOT Error (java.lang.ClassNotFoundException)
Message     class sorcerersisle.cfpassphrase.lucee.PassphraseHash is invalid or doesn't exist
Cause   java.lang.ClassNotFoundException
Stacktrace  The Error Occurred in
D:\Inetpub\wwwroot\xxxx\indexframe.cfm: line 44

```
42: <cfquery name="UpdateneuesPW" datasource="factum_awbs">
43: Update xxxx_User
44: set PassWordNEU= '#PassphraseHash( Form.Password )#'
45: WHERE (UserID = '#CheckUser.Userid#') AND (PasswordNEU = '' OR PasswordNEU IS NULL)
46: </cfquery>
```

Java Stacktrace     lucee.runtime.exp.NativeException: class sorcerersisle.cfpassphrase.lucee.PassphraseHash is invalid or doesn't exist
    at lucee.commons.lang.PhysicalClassLoader.findClass(PhysicalClassLoader.java:146)
    at lucee.commons.lang.PhysicalClassLoader.loadClass(PhysicalClassLoader.java:127)
    at lucee.commons.lang.PhysicalClassLoader.loadClass(PhysicalClassLoader.java:107)
    at hepatitis_c_register990.indexframe_cfm$cf.call(D:\Inetpub\wwwroot\Hepatitis-C-Register\indexframe.cfm:44)
    at lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:897)
    at lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:812)
    at lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:807)
    at lucee.runtime.listener.ClassicAppListener._onRequest(ClassicAppListener.java:62)
    at lucee.runtime.listener.MixedAppListener.onRequest(MixedAppListener.java:44)
    at lucee.runtime.PageContextImpl.execute(PageContextImpl.java:2256)
    at lucee.runtime.PageContextImpl._execute(PageContextImpl.java:2248)
    at lucee.runtime.PageContextImpl.executeCFML(PageContextImpl.java:2216)
    at lucee.runtime.engine.CFMLEngineImpl.serviceCFML(CFMLEngineImpl.java:882)
    at lucee.loader.engine.CFMLEngineWrapper.serviceCFML(CFMLEngineWrapper.java:103)
    at lucee.loader.servlet.CFMLServlet.service(CFMLServlet.java:62)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:528)
    at org.apache.coyote.ajp.AbstractAjpProcessor.process(AbstractAjpProcessor.java:872)
    at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:672)
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520)
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.lang.Thread.run(Thread.java:745)
 Caused by: java.lang.ClassNotFoundException: class sorcerersisle.cfpassphrase.lucee.PassphraseHash is invalid or doesn't exist
    ... 36 more

Timestamp   20.07.16 18:49:48 MESZ

System: Windows Server 2012 R2 (6.3) 64bit
Servlet Container   Apache Tomcat/8.0.35
Java 1.8.0_92 (Oracle Corporation) 64bit  
Architecture 64bit  

Anyone any ideas?
All hints/tipps are welcome!

Thank you
Michael

**Comment by Meikelv** [comment]
After understanding github (:-)) I found another comment from andrew dixon who found the cause of this problem.
It would be nice, if you could provide and osgi-File of cfpassphrase on your website (I tried it, but I have no success, so I hope, someone could handle it and provide the file).

**Comment by Meikelv** [comment]
Hi,
now it is working. Andrew helped me to solve it:
1. I'm using Lucee 5.1.0.17-BETA
2. Andrew sent me the osgi-Verson of cfpassphrase (thank you very much!!!)
3. I modified the fld-file like Andrew described it here: https://github.com/boughtonp/cfpassphrase/issues/18

Bye
Michael

Maybe, you could add this files to your repository?

**Comment by Meikelv** [comment]
Now I'm testing Lucee 5.2.0.4-snapshot and I'm getting agani a java.lang.NullpointerException.
So I reopen the task. 
Let's see, if there is a solution :-)

**Comment by Meikelv** [comment]
Don't know why and how, but I reinstalled lucee, now with newer snapshot (5.2.0.4) and copied the osgi-bundle (cfpassphrase-bundle.jar) to c:\lucee\tomcat\lucee-server\bundles\ and the fld after editing (<class> to <class bundle-name="sorcerersisle.cfpassphrase" bundle-version="0.1">) into c:\lucee\tomcat\lucee-server\context\library\fld\ and the tld to c:\lucee\tomcat\lucee-server\context\library\tld\ and now **it works!!!**

Thank you very much!

**Comment by andrew-dixon** [comment]
Use the extension version I've created, which you can download here and install via the admin:

https://s3-eu-west-1.amazonaws.com/cfpassphrase/cfpassphrase.lex

**Comment by boughtonp** [comment]
This should at last be addressed by the latest 0.2 release candidate, which has been tested against a fresh Lucee Express 5.3.6.53-RC.

Available in both JAR and Lucee LEX format [here](https://www.sorcerersisle.com/software/cfpassphrase).

**Comment by boughtonp** [comment]
Release of cfPassphrase v0.2 includes LEX
